src/TLuaInterpreter.cpp

```diff
@@ -11728,35 +11728,264 @@ int TLuaInterpreter::sendSocket( lua_State * L )
     return 0;
 }
 
+/** sendIrc( Target, Message )
+ *  Sends a Message to given Target.
+ *  Returns true or false if the message was able to be sent.
+ */
 int TLuaInterpreter::sendIrc( lua_State * L )
 {
-    string who;
-    if( ! lua_isstring( L, 1 ) )
-    {
-        lua_pushstring( L, "sendIrc: wrong argument type" );
-        lua_error( L );
-        return 1;
-    }
-    else
-    {
+    string who, text;
+    if( ! lua_isstring( L, 1 ) ) {
+        lua_pushfstring(L, "sendIrc: bad argument #1 type (Target as string expected, got %s!)", lua_typename(L, 1));
+        return lua_error(L);
+    } else {
         who = lua_tostring( L, 1 );
     }
-    string text;
-    if( ! lua_isstring( L, 2 ) )
-    {
-        lua_pushstring( L, "sendIrc: wrong argument type" );
-        lua_error( L );
-        return 1;
-    }
-    else
-    {
+    if( ! lua_isstring( L, 2 ) ) {
+        lua_pushfstring(L, "sendIrc: bad argument #2 type (Message as string expected, got %s!)", lua_typename(L, 2));
+        return lua_error(L);
+    } else {
         text = lua_tostring( L, 2 );
     }
-    QString chan = who.c_str();
-    QString txt = text.c_str();
-    if( ! mudlet::self()->mpIRC ) return 0;
-    mudlet::self()->mpIRC->connection->sendCommand( IrcCommand::createMessage( chan, txt ) );
-    return 0;
+    QString target = who.c_str();
+    QString msg = text.c_str();
+    Host* pHost = &getHostFromLua(L);
+    if ( !mudlet::self()->mpIrcClientMap[pHost] ) {
+        // create a new irc client if one isn't ready.
+        mudlet::self()->mpIrcClientMap[pHost] = new dlgIRC(pHost);
+        mudlet::self()->mpIrcClientMap[pHost]->raise();
+        mudlet::self()->mpIrcClientMap[pHost]->show();
+    }
+
+    // wait for our client to be ready before sending messages.
+    if (!mudlet::self()->mpIrcClientMap[pHost]->mReadyForSending) {
+        lua_pushboolean(L, false);
```

Comment:
Could you return nil+error instead, just like you did with `connectToServer()`? Just a `false` will be frustrating when something is not working and you have no idea why.